src/module_wrap.cc

```diff
@@ -667,14 +725,49 @@ void ModuleWrap::Resolve(const FunctionCallbackInfo<Value>& args) {
     return;
   }
 
-  Maybe<URL> result = node::loader::Resolve(env, specifier_std, url, true);
-  if (result.IsNothing() || (result.FromJust().flags() & URL_FLAGS_FAILED)) {
+  if (!args[2]->IsBoolean()) {
+    env->ThrowError("third argument is not a boolean");
+    return;
+  }
+  bool set_package_esm = args[2]->ToBoolean().As<v8::Boolean>()->Value();
+
+  Maybe<URL> result = node::loader::Resolve(env, specifier_std, url);
+  if (result.IsNothing() ||
+      (result.FromJust().flags() & URL_FLAGS_FAILED)) {
     std::string msg = "Cannot find module " + specifier_std;
     env->ThrowError(msg.c_str());
     return;
   }
 
-  args.GetReturnValue().Set(result.FromJust().ToObject(env));
+  bool esm = false;
+  if (set_package_esm) {
+    SetPackageModeEsm(env, result.FromJust());
+    esm = true;
+  } else {
+    std::string filePath = result.FromJust().ToFilePath();
+    // check the package esm mode for ambiguous extensions
+    if (filePath.substr(filePath.length() - 3, 3) != ".mjs" &&
+        filePath.substr(filePath.length() - 5, 5) != ".json" &&
+        filePath.substr(filePath.length() - 5, 5) != ".node") {
+      esm = GetPackageModeEsm(env, result.FromJust());
```

Comment:
Possible typo, I noticed the ".json" and ".node" offsets were 5 and they have 5 characters in their strings but the ".mjs" offset is 3 but it has 4 characters in its string.

Reply:
👍 corrected.